globus_automate_client/cli/callbacks.py

```diff
@@ -37,31 +46,32 @@ def _base_principal_validator(
     Globus ID beginnings. It will optionally determine if a provided principal
     exists in a set of "special" values.
     """
-    groups_beginning = "urn:globus:groups:id:"
     auth_beginning = "urn:globus:auth:identity:"
 
+    auth_client: Optional[AuthClient] = None
+
+    valid_principals = []
+
+    invalid_principals = []
     for p in principals:
-        if special_vals and p in special_vals:
-            continue
-
-        valid_beggining = False
-        for beggining in [groups_beginning, auth_beginning]:
-            if p.startswith(beggining):
-                uuid = p[len(beggining) :]
-                try:
-                    UUID(uuid, version=4)
-                except ValueError:
-                    raise typer.BadParameter(
-                        f"Principal could not be parsed as a valid identifier: {p}"
-                    )
-                else:
-                    valid_beggining = True
-        if not valid_beggining:
-            raise typer.BadParameter(
-                f"Principal could not be parsed as a valid identifier: {p}"
-            )
-
-    return principals
+        if special_vals and p in special_vals or re.match(_principal_urn_regex, p):
+            valid_principals.append(p)
+        else:  # Try to do a lookup of the identity
+            if auth_client is None:
+                auth = get_authorizer_for_scope(
+                    "urn:globus:auth:scope:auth.globus.org:view_identities"
+                )
+                auth_client = AuthClient(authorizer=auth)
+            auth_resp = auth_client.get_identities(usernames=p)
```

Comment:
Will this raise a GlobusAPIError if p isn't found? If so I think we can catch that here to keep this from crashing.

As a side note, will this end up getting all linked identities for a principal `p`? If so I'm wary of doing that look up. I can see a user specifying an identity on --manage-by, and in this implementation all of the user's linked identities would be passed onto the manage_by field of an Action/Run, which feels like a client side version of the the work we did server side to prevent linked identities being added by default.

Reply:
Turns out this doesn't throw an error if the identity doesn't exist, so we should make sure it does do that.

Reply:
fixed!!